lib/filterx/object-datetime.c

```diff
@@ -205,6 +205,38 @@ _repr(FilterXObject *s, GString *repr)
   return datetime_repr(&ut, repr);
 }
 
+static FilterXObject *
+_add(FilterXObject *lhs_object, FilterXObject *rhs_object)
```

Comment:
This is a virtual method for adding an other object to `self`, I think the signature should represent that.
```c
_add(FilterXObject *self, FilterXObject *object)
```
Changing the signature would make it more natural to see that `lhs_object` / `self` is always `datetime` typed. This applies to the rest of the types' add(), too.